mdoc/src/main/scala-3/mdoc/internal/markdown/MarkdownCompiler.scala

```diff
@@ -0,0 +1,226 @@
+package mdoc.internal.markdown
+
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.PrintStream
+import java.net.URL
+import java.net.URLClassLoader
+import java.nio.file.Path
+import java.nio.file.Paths
+import mdoc.Reporter
+import mdoc.document.Document
+import mdoc.document._
+import mdoc.internal.document.DocumentBuilder
+import mdoc.internal.document.MdocNonFatal
+import mdoc.internal.pos.TokenEditDistance
+import scala.collection.JavaConverters._
+import scala.collection.Seq
+import scala.meta.Classpath
+import scala.meta.AbsolutePath
+import scala.meta.inputs.Input
+import scala.meta.inputs.Position
+
+import scala.meta.internal.inputs.XtensionInputSyntaxStructure
+
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.interactive.Interactive
+import dotty.tools.dotc.interactive.InteractiveCompiler
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.interfaces.SourcePosition
+import dotty.tools.dotc.ast.Trees.Tree
+import dotty.tools.dotc.interfaces.{SourceFile => ISourceFile}
+import dotty.tools.dotc.reporting.Diagnostic
+import dotty.tools.dotc.parsing.Parsers.Parser
+import dotty.tools.dotc.Compiler
+import dotty.tools.io.{AbstractFile, VirtualDirectory}
+import dotty.tools.repl.AbstractFileClassLoader
+import dotty.tools.dotc.util.SourceFile
+import java.io.File
+import java.net.URI
+import java.net.URL
+import java.net.URLClassLoader
+import sun.misc.Unsafe
+import scala.annotation.implicitNotFound
+import dotty.tools.dotc.config.Settings.Setting._
+import mdoc.internal.CompatClassloader
+import mdoc.internal.pos.PositionSyntax._
+
+class MarkdownCompiler(
```

Comment:
Main entry point to the Scala 3 compiler. Analogous class for Scala 2 and moved out `MarkdownCompiler` object and renamed to `MarkdownBuilder` - it's cross compiled now.